Mathport/Syntax/Translate/Basic.lean

```diff
@@ -157,8 +171,32 @@ elab:max "warn!" interpStr:interpolatedStr(term) or:(checkColGt "|" term)? : ter
     logComment str
     $or:term)
 
-def trExpr (e : Expr) : M Syntax := do (← read).trExpr e
-def trCommand (e : Command) : M Unit := do (← read).trCommand e
+def positionToStringPos (pos : Position) : String.Pos :=
+  10000 * pos.line + pos.column -- moderately hacky
+
+def stringPosToLine (pos : String.Pos) : Nat :=
+  pos / 10000 -- slightly more hacky
+
+def withSpan (m : Option Meta) (k : M α) : M α :=
+  match m with
+  | none => k
+  | some { start, end_, .. } =>
+    let sourceInfo := SourceInfo.synthetic (positionToStringPos start) (positionToStringPos end_)
+    withRef (Syntax.atom sourceInfo "") k
+
+def unspanning (k : β → M α) (x : Spanned β) : M α := withSpan x.meta do k x.kind
+def unspanningS (k : β → M Syntax) (x : Spanned β) : M Syntax := do
```

Comment:
I got a bit confused by the name here; perhaps `spanning(S)` would be better since it is adding a span to the result of the provided (unspanned) function

Reply:
Okay, I don't really care about the name.